### PR TITLE
[native] Fix TaskManagerTest.buildSpillDirectoryFailure

### PR DESCRIPTION
**Pull request by spershin**
## Description
There is a race in the test itself, which we fix here:
- taskManager_->cleanOldTasks() can be called before the Driver is destroyed.
- Since the Driver holds the Task, the Task is considered a zombie inside cleanOldTasks() and is NOT removed from the map.
- So, destructor of the Task is not called.
- Then we call waitForAllTasksToBeDeleted() which simply compares how many ctors and dtors we had called. They don't match.
- Test fails.

Fixes https://github.com/prestodb/presto/issues/23062

## Test Plan
Stress testing.
`buck2 test @mode/opt github/presto-trunk/presto-native-execution/presto_cpp/main/tests:presto_main_test -- TaskManagerTest.buildSpillDirectoryFailure --stress-runs 100
`

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by tanjialiang** [comment]
Thanks for the fix. Does it make sense to just extend the test from velox's OperatorTestBase and reuse the rich utilities there?

**Comment by spershin** [comment]
@tanjialiang 
This test fixes the use of TaskManager's cleanOldTasks(). TaskManager is not in Velox, so we cannot use rich utilities from Velox.

File: presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp [hunk 1]
```diff
@@ -54,6 +54,26 @@ namespace facebook::presto {
 
 namespace {
 
+// Repeatedly calls for cleanOldTasks() for a while to ensure that we overcome a
```
Comment (elharo) [inline presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp:57]: for cleanOldTasks() -->= cleanOldTasks()

**Review by elharo — APPROVED** [review]

**Review by amitkdutta — APPROVED** [review]
Thanks for the fix @spershin

File: presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp (hunk 1 above)
Comment (spershin) [reply presto-native-execution/presto_cpp/main/tests/TaskManagerTest.cpp:57]: Thanks!
Let's fix this some other time.
Now it is important to land this to remove the noise.

**Pull request merged**